contracts-periphery/src/test/UmbraBatchSend.t.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "./utils/DSTestPlus.sol";
+import "../UmbraBatchSend.sol";
+
+interface UmbraToll {
+  function toll() external returns(uint256);
+}
+
+contract UmbraBatchSendTest is DSTestPlus {
+  using stdStorage for StdStorage;
+  StdStorage stdstore;
+
+  address umbra;
+  UmbraBatchSend router;
+  MockERC20 token;
+
+  address tokenAddr;
+  address alice = address(0x202204);
+  address bob = address(0x202205);
+
+  uint256 alicePrevBal;
+  uint256 bobPrevBal;
+  uint256 umbraPrevBal;
+
+  uint256 toll;
+  bytes32 pkx = "pkx";
+  bytes32 ciphertext = "ciphertext";   
+
+  UmbraBatchSend.SendEth[] sendEth;
+  UmbraBatchSend.SendToken[] sendToken;
+
+  error ValueMismatch();
+
+  function setUp() virtual public {
+    umbra = deployCode("src/test/utils/Umbra.json", bytes(abi.encode(0, address(this), address(this))));
+    vm.etch(0xFb2dc580Eed955B528407b4d36FfaFe3da685401, umbra.code);
+    umbra = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
```

Comment:
This can be a constant that we declare on L15, then we can use that value in the `etch` call on L38

Reply:
Looks like now we declare `address umbra` on L13 and `address constant umbraAddr` on L20, and use both in the tests. We only need one of these—I'd stick with with `address constant umbra = 0x123` and delete `umbraAddr`